### PR TITLE
(DOCSP-22334) Adds reference section and cluster configuration file info

### PR DESCRIPTION
**Pull request by sarahsimpers**
[Build Link](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=62fbb614b354ebcf05ed397c)
[Jira Ticket Link](https://jira.mongodb.org/browse/DOCSP-22334)
[Staging](https://docs-mongodbcom-staging.corp.mongodb.com/atlas-cli/docsworker-xlarge/DOCSP-22334/cluster-config-file/)

I tested and confirmed that the JSON examples both work. The YAML examples were removed.

### CODE REVIEW

**Review by andreaangiolillo — COMMENTED** [review]
Could you remove the YAML example? It is not possible to use a YAML file for a cluster config file at this time, I will create a ticket to add this functionality. The other changes LGTM

File: source/cluster-config-file.txt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+   :widths: 20 10 70 
+
+   * - Field 
+     - Type 
+     - Description 
+
+   * - ``clusterType``
+     - string
+     - Type of {+cluster+} that you want to create. Valid values are
+       ``REPLICASET``, ``SHARDED``, or ``GEOSHARDED``.
```
Comment (corryroot) [inline source/cluster-config-file.txt:44]:
```suggestion
     - Human-readable label that indicates the type of {+cluster+} to create.
       Values include: ``REPLICASET``, ``SHARDED``, or ``GEOSHARDED``.
```

File: source/cluster-config-file.txt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+   * - ``name``
+     - string
+     - Human-readable label that identifies the {+cluster+}.
+
+   * - ``replicationSpecs``
+     - array
+     - List containing the configuration settings for {+cluster+}
+       regions and the hardware provisioned in them.
```
Comment (corryroot) [inline source/cluster-config-file.txt:53]: **Style:** See [Clarify Gerunds and Participles](https://www.mongodb.com/docs/meta/style-guide/writing/clarify-gerunds-participles/)
```suggestion
     - List that contains the configuration settings for your {+cluster+}
       regions and the hardware provisioned in them.
```

File: source/cluster-config-file.txt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+       MongoDB nodes that |service| deploys to the region.
+
+       Each regionConfigs object must have either an ``analyticsSpecs``
+       object, ``electableSpecs`` object, or ``readOnlySpecs`` object.
+
+       - Shared {+clusters+} require only ``electableSpecs``.
+       - Dedicated {+clusters+} can specify any of these
+         specifications, but must have at least one ``electableSpecs`` object within a ``replicationSpec``.
+       - Every hardware specification must use the same
+         ``instanceSize``.
```
Comment (corryroot) [inline source/cluster-config-file.txt:70]: **Style:** See [Clarify Gerunds and Participles]. Line break.
```suggestion
     - List that contains the hardware specifications for the nodes set within
       the region that you specify. Each ``regionConfigs`` object describes
       the region's priority in elections and the number and type of
       MongoDB nodes that |service| deploys to the region.

       Each regionConfigs object must have either an ``analyticsSpecs``
       object, ``electableSpecs`` object, or ``readOnlySpecs`` object.

       - Shared {+clusters+} require only ``electableSpecs``.
       - Dedicated {+clusters+} can specify any of these
         specifications, but must have at least one ``electableSpecs``
         object within a ``replicationSpec``.
       - Every hardware specification must use the same
         ``instanceSize``.
```

File: source/cluster-config-file.txt [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+   * - ``replicationSpecs.``
+       ``regionConfigs.electableSpecs``
+     - object
+     - Hardware specifications for electable nodes in the region.
+       Electable nodes can become the primary and can enable local
```
Comment (corryroot) [inline source/cluster-config-file.txt:76]: **Suggestion:** Link to term.

```suggestion
       Electable nodes can become the :term:`primary` and can enable local
```

File: source/cluster-config-file.txt [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+       replicationSpecs[n].regionConfigs[m] objects must total 3, 5, or
+       7.
+
+   * - ``replicationSpecs.``
+       ``regionConfigs.priority``
+     - Integer
+     - Precedence that is given to this region when a primary election
+       occurs.
+
+       If your region has set ``electableSpecs.nodeCount`` to ``1`` or higher, it must have a priority of exactly one (1) less than another region in the ``replicationSpecs[n].regionConfigs[m]`` array. The highest-priority region must have a priority of ``7``. The lowest possible priority is ``1``.
```
Comment (corryroot) [inline source/cluster-config-file.txt:104]: **Nit:** Missing line breaks.

```suggestion
       If your region has set ``electableSpecs.nodeCount`` to ``1`` or higher, it 
       must have a priority of exactly one (1) less than another region in the 
       ``replicationSpecs[n].regionConfigs[m]`` array. The highest-priority region must have 
       a priority of ``7``. The lowest possible priority is ``1``.
```

File: source/cluster-config-file.txt [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+       The priority ``7`` region identifies the preferred region of the
+       {+cluster+}. |service| places the primary node in the preferred region.
+
+   * - ``replicationSpecs.``
+       ``regionConfigs.providerName``
+     - string
+     - Name of your cloud service provider. Valid values are ``AWS``,
+       ``AZURE``, or ``GCP``.
```
Comment (corryroot) [inline source/cluster-config-file.txt:113]:
```suggestion
     - Human-readable label that identifies your cloud service provider.
       Values include: ``AWS``, ``AZURE``, or ``GCP``.
```

File: source/cluster-config-file.txt [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+       To create a multi-cloud {+cluster+}, specify more than one
+       service provider for your ``regionConfigs`` objects. To
+       learn more, see the :ref:`multi-cloud example configuration file
+       <multi-cloud-example-cluster-config-file>`
+
+   * - ``replicationSpecs.``
+       ``regionConfigs.regionName``
+     - string
+     - Physical location of your {+cluster+} nodes. The region you
```
Comment (corryroot) [inline source/cluster-config-file.txt:123]:
```suggestion
     - Human-readable label that indicates the physical location of your {+cluster+}
     nodes. The region you
```

File: source/cluster-config-file.txt [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+       choose can affect network latency for clients accessing your
+       databases.
+     
+       For a complete
+       list of supported |aws| regions, see `AWS regions <https://docs.atlas.mongodb.com/reference/amazon-aws/#amazon-aws>`__. 
+       
+       For a complete list of supported |azure| regions, see `Azure regions <https://docs.atlas.mongodb.com/reference/microsoft-azure/#microsoft-azure>`__. 
+       
+       For a complete list of supported |gcp| regions, see `Google Cloud regions <https://docs.atlas.mongodb.com/reference/google-gcp/#google-gcp>`__.
```
Comment (corryroot) [inline source/cluster-config-file.txt:132]: **Suggestion:** Simplify and avoid redundancy. Use the `:atlas:` role.

```suggestion
   For a complete list of region name values, refer to the
   the cloud provider reference pages:

   - :atlas:`AWS </reference/amazon-aws/#amazon-aws>>`

   - :atlas:`GCP </reference/google-gcp/#google-gcp>`

   - :atlas:`Azure </reference/microsoft-azure/#microsoft-azure>`

```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: source/cluster-config-file.txt [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+
+.. _optional-cluster-settings:
+
+Optional and Conditional {+Cluster+} Settings
+---------------------------------------------
+
+Your {+cluster+} configuration file may contain additional optional or conditional {+cluster+} settings. For a full list of available
+settings, see the API documentation for 
+`Create One Multi-Cloud Cluster from One Project 
+<https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#operation/createOneAdvancedClusterFromOneProject>`__.
```
Comment (corryroot) [inline source/cluster-config-file.txt:142]: **Suggestions:** Line breaks. Use the `:atlas:` role.

```suggestion
Your {+cluster+} configuration file may contain additional optional or conditional 
{+cluster+} settings. For a full list of available settings, see the API documentation for 
:atlas:`Create One Multi-Cloud Cluster from One Project 
</reference/api-resources-spec/#operation/createOneAdvancedClusterFromOneProject>`.
```

File: source/cluster-config-file.txt [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+.. literalinclude:: /includes/json-cluster-config-file.json
+
+.. _multi-cloud-example-cluster-config-file:
+
+Multi-Cloud Example Cluster Configuration File
+----------------------------------------------
+
+To create a multi-cloud {+cluster+}, specify more than one
+service provider for your ``regionConfigs`` objects as shown in the following example file:
```
Comment (corryroot) [inline source/cluster-config-file.txt:159]: **Nit:** Line breaks.

```suggestion
To create a {+cluster+} using a single cloud provider, specify the same service 
provider for your ``regionConfigs`` objects as shown in the following example file:

.. literalinclude:: /includes/json-cluster-config-file.json

.. _multi-cloud-example-cluster-config-file:

Multi-Cloud Example Cluster Configuration File
----------------------------------------------

To create a multi-cloud {+cluster+}, specify more than one
service provider for your ``regionConfigs`` objects as shown in the following example 
file:
```

**Review by corryroot — COMMENTED** [review]
Thanks Sarah! See comments.

File: source/cluster-config-file.txt [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+   * - ``replicationSpecs.``
+       ``regionConfigs``
+     - array
+     - List that contains the hardware specifications for the nodes set
+       within the region that you specify. Each ``regionConfigs``
+       object describes the region's priority in elections and the
+       number and type of MongoDB nodes that |service| deploys to the
+       region.
+       
+       Each regionConfigs object must have either an ``analyticsSpecs``
```
Comment (corryroot) [inline source/cluster-config-file.txt:66]:
```suggestion
       Each ``regionConfigs`` object must have either an ``analyticsSpecs``
```

**Review by corryroot — APPROVED** [review]
LGTM! I added one nit.

**Pull request merged**